packages/gasket-request/lib/request.js

```diff
@@ -0,0 +1,70 @@
+const cache = new WeakMap();
+
+/**
+ * @type {import('./index.js').objectFromCookieStore}
+ */
+async function objectFromCookieStore(cookieStore) {
+  return (await cookieStore.getAll())
+    .reduce((acc, { name, value }) => {
+      acc[name] = value;
+      return acc;
+    }, {});
+}
+
+/**
+ * @type {import('./index.js').GasketRequest}
+ */
+export class GasketRequest {
+  constructor(normalizedRequest) {
+    this.headers = normalizedRequest.headers;
+    this.cookies = normalizedRequest.cookies;
+    this.query = normalizedRequest.query;
+    this.path = normalizedRequest.path;
+  }
+}
+
+/**
+ * @type {import('./index.js').makeGasketRequest}
+ */
+export async function makeGasketRequest(requestLike) {
+  if (requestLike instanceof GasketRequest) {
+    return requestLike;
+  }
+
+  const { headers, cookies = {} } = requestLike;
+
+  if (!headers) {
+    throw new Error('request argument must have headers');
+  }
+
+  if (cache.has(headers)) {
+    return cache.get(headers);
+  }
+
+  const normalize = async () => {
+    let query = requestLike.query;
+    let path = requestLike.path;
```

Comment:
This property does not exist on [IncomingMessage](https://nodejs.org/docs/latest-v22.x/api/http.html#class-httpincomingmessage). There is a [`url` property](https://nodejs.org/docs/latest-v22.x/api/http.html#messageurl), however. Maybe we could have a parsed URL as part of the normalization to cover both path and query.